src/Nat.mo

```diff
@@ -386,6 +386,63 @@ module {
     }
   };
 
+  /// Returns an iterator over `Nat` values from the first to second argument with an exclusive upper bound,
+  /// incrementing by the specified step size. The step can be positive or negative.
+  /// ```motoko include=import
+  /// import Iter "mo:base/Iter";
+  ///
+  /// // Positive step
+  /// let iter1 = Nat.step(1, 7, 2);
+  /// assert(?1 == iter1.next());
+  /// assert(?3 == iter1.next());
+  /// assert(?5 == iter1.next());
+  /// assert(null == iter1.next());
+  ///
+  /// // Negative step
+  /// let iter2 = Nat.step(7, 1, -2);
+  /// assert(?7 == iter2.next());
+  /// assert(?5 == iter2.next());
+  /// assert(?3 == iter2.next());
+  /// assert(null == iter2.next());
+  /// ```
+  ///
+  /// If `step` is 0 or if the iteration would not progress towards the bound, returns an empty iterator.
+  public func step(fromInclusive : Nat, toExclusive : Nat, step : Int) : Iter.Iter<Nat> {
+    if (step == 0) {
+      Iter.empty()
```

Comment:
I think Python actually traps on step == 0, if we are trying to match Python. Not sure what is does about lack of progress.

Reply:
That's correct; Python raises an error when step is 0. 

I suppose we could remain fully consistent, although it seems a bit odd to trap for this but not when the lower bound is greater than the upper bound. Maybe we could trap for both conditions to reduce the chance of unintended bugs, even though this is a bit stricter than the behavior in Python?